ext/IntervalArithmeticForwardDiffExt.jl

```diff
@@ -56,4 +56,22 @@ function Base.:(^)(x::ExactReal, y::Dual{<:Any, I}) where I<:Interval
     return convert(I, x)^y
 end
 
+function Base.:(^)(x::Dual{<:Tx}, y::ExactReal) where Tx
+    v = value(x)
+    expv = v^y
+    if iszero(y) || all(iszero.(values(partials(x))))
+        new_partials = zero(partials(x))
+    else
+        new_partials = partials(x) * y * v^(y - 1)
```

Comment:
I have not tried this PR yet, but just looking at the code I would have expected that we need `v^(y - interval(1))`.
In which case, we should have a test that reflects this (maybe if `y = ExactReal(pi)` this is relevant?)

Reply:
The interval functions were previously implemented and work fine. The two new functions are for mixing `ExactReal` with non-interval `Real` numbers, so no conversion to interval should be added.

The default ForwardDiff implementation uses `x == zero(x)` to check the exponent, and is thus failing with `ExactReal`s, just like it was failing with `Interval`s.

Reply:
You are right, but `ExactReal` should propagate though, no?
Otherwise `v^(y - 1)` would give a NG flag despite `v` being an interval and `y` an ExactReal; maybe `new_partials = partials(x) * y * v^(y - convert(typeof(v), 1))` (or something in this vein)?

Reply:
If you mix intervals and `ExactReal`s, you hit the definition at lines 51-57, and the `ExactReal` is immediately converted to an interval.

The new functions have nothing to do with intervals, it is a fix around the fact that ForwardDiff decided to use branching in the derivatives of powers, which also breaks `ExactReal`s.